core/app/models/spree/promotion_handler/coupon.rb

```diff
@@ -25,6 +25,20 @@ def apply
         self
       end
 
+      def remove
+        if !coupon_code.present? || !promotion.present?
```

Comment:
As `present?` is `!blank?` (see [here](https://apidock.com/rails/Object/present%3F)) then  this can become `if coupon_code.blank? || promotion.blank?`